packages/core/strapi/src/cli/commands/cloud/utils/token.ts

```diff
@@ -0,0 +1,138 @@
+import jwksClient, { type JwksClient, type SigningKey } from 'jwks-rsa';
+import type { JwtHeader, VerifyErrors } from 'jsonwebtoken';
+import jwt from 'jsonwebtoken';
+import fs from 'fs';
+import path from 'path';
+import { getStoragePath, TMP_TOKEN_FILE } from '../config/local';
+import type { CloudCliConfig } from '../types';
+import { cloudApiFactory } from '../services/cli-api';
+import type { CLIContext } from '../../../types';
+
+const cloudApiService = cloudApiFactory();
+
+let cliConfig: CloudCliConfig;
+
+interface DecodedToken {
+  [key: string]: any;
+}
+
+export function tokenServiceFactory({ logger }: CLIContext) {
+  function saveToken(str: string) {
+    const storagePath = getStoragePath();
+    const filePath = path.join(storagePath, TMP_TOKEN_FILE);
```

Comment:
It makes me nervous to just store the token directly as a file. Could we use something like [keytar](https://github.com/atom/node-keytar) to make use of the system's keychain? I believe that's how most cli logins such as github work.

Reply:
I understand your concern, but given that our source code is open I don't think this will completely mitigate the risk of token exposure. While using something as `keytar` might add some additional layer of security this can come at the expense of introducing a new dependency, which increases complexity and maintenance overhead. FYI @nathan-pichon

Reply:
Even if I do think security is important, tokens are already coming with some limitations like expirability.
Also, users that wants an increased amount of security may opt for git repo hosting directly on cloud web interface.
In a further iteration, we want to include refresh token and short live tokens. But it will not be implemented in the mvp.